src/satosa/base.py

```diff
@@ -134,6 +134,12 @@ def _auth_resp_callback_func(self, context, internal_response):
         """
 
         context.request = None
+        context_state = context.state.get(STATE_KEY)
+        if not context_state:
```

Comment:
Which are the conditions that make this possible? Why would the state be empty?

Reply:
This is a superior knowledge that I would gain when I grow up ;)

That's the footprint I collected:
https://github.com/IdentityPython/SATOSA/pull/326#issuecomment-620572473

Which method do you suggest to do, should we pick all the things with a python debugger or we should move to a big view that let us to satisfy our users without make too much noise in the code. It's a kind of paternalism that we acquire getting older, the users are tring to tell us something ... 

Probably you know better  than me how and when the context.state became inconsistent. Let's try to talk about this is a separate thread if you agree